src/Cli/dotnet/Parser.cs

```diff
@@ -178,17 +178,23 @@ internal static int ExceptionHandler(Exception exception, ParseResult parseResul
 
             if (exception is Utils.GracefulException)
             {
-                Reporter.Error.WriteLine(CommandLoggingContext.IsVerbose
-                    ? exception.ToString().Red().Bold()
-                    : exception.Message.Red().Bold());
+                Reporter.Error.WriteLine(CommandLoggingContext.IsVerbose ?
+                    exception.ToString().Red().Bold() :
+                    exception.Message.Red().Bold());
             }
             else if (exception is CommandParsingException)
             {
-                Reporter.Error.WriteLine(CommandLoggingContext.IsVerbose
-                    ? exception.ToString().Red().Bold()
-                    : exception.Message.Red().Bold());
+                Reporter.Error.WriteLine(CommandLoggingContext.IsVerbose ?
```

Comment:
should the reporter maybe be initialized with a commandloggingcontext, and abstract the logic away so when you log an exception, it does the check to decide to log the full exception or just the message?

Reply:
Reporter.Error is an IReporter that gets statically initialized...CommandLoggingContext.IsVerbose is also statically initialized, but I'm not sure if there's an explicit order between the two, so that might turn it into a race condition.

It would also mean I'd have to add a WriteException method to IReporter and all its implementations, and it only really makes sense for this one.